### PR TITLE
task 7 done

### PR DESCRIPTION
**Pull request by kellysum**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -94,8 +95,95 @@ describe("GET /api",()=>{
...
+        .post('/api/articles/1/comments')
+        .send(newComment)
+        .expect(201)
+        .then(({body})=>{
+            const {postedComments} = body
+            expect(postedComments.article_id).toBe(1)
+            expect(postedComments.body).toBe(newComment.comment)
+        })
+    })
+    test('400: responds with an error message if the article_id does not exist', ()=>{
```
Comment (nc-jonathan-mcguire) [inline __tests__/app.test.js]: should be article id is invalid

File: __tests__/app.test.js [hunk 2]
```diff
@@ -94,8 +95,95 @@ describe("GET /api",()=>{
...
+        const newComment = {username : 'butter_bridge', comment : 'testing comment'}
+        return request(app)
+        .post('/api/articles/notnumber/comments')
+        .send(newComment)
+        .expect(400)
+        .then(({body})=>{
+            expect(body.msg).toBe("Bad request")
+        })
+    })
+    test('400: responds with an error message if the username does not exist', ()=>{
```
Comment (nc-jonathan-mcguire) [inline __tests__/app.test.js]: would be 404 - username not found - the request is valid, its just not found

File: controller/controller.js [hunk 3]
```diff
@@ -20,7 +21,24 @@ exports.getArticleId = (req, res, next)=>{
...
+    .then((articles)=>{
+        res.status(200).send({articles})
+    })
+    .catch(next)
+}
+exports.postComment = (req, res, next)=>{
+    const {article_id} = req.params 
+   
+    const newComment = req.body
+    insertComment(newComment, article_id).then((postedComments)=>{
```
Comment (nc-jonathan-mcguire) [inline controller/controller.js]: shoiuld only be one comment, so makes sense to call the key `comment`

File: model/comments.model.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+exports.insertComment=(comments, article_id) =>{
+    const {username, comment} = comments
+    console.log(username, comment)
+    const psqlQuery = `INSERT INTO comments (author, body, article_id) VALUES ($1, $2, $3) RETURNING *;`
+
+    return db.query(psqlQuery, [username, comment, article_id])
+    .then((data)=>{
+       console.log(data.rows[0])
```
Comment (nc-jonathan-mcguire) [inline model/comments.model.js]: remove log

**Review by nc-jonathan-mcguire — CHANGES_REQUESTED** [review]
Just a few changes to make please and a test to add:

- [ ] Status 400: Bad request, incomplete body - Request body incomplete/missing required field(s) of username or body.

Please see my comments for other changes needed

**Review by ghost — APPROVED** [review]
Nicely done. Merge away!

**Pull request merged**